tinytuya/core.py

```diff
@@ -109,6 +124,7 @@
 DEVICEFILE = 'devices.json'
 RAWFILE = 'tuya-raw.json'
 SNAPSHOTFILE = 'snapshot.json'
+DPMAPPINGSFILE = 'mappings.json'

```

Comment:
Is there a reason we want this as a separate file instead of embedding the map directly into devices.json?  I'm fine with as-designed, just making sure we aren't missing something.

Reply:
We definitely can, I just had initially approached this as an optional add-on and not a core part.  If we're going to default to 'yes' then I think it would make more sense to just roll this into the device list downloader instead of calling it as a separate step.

Reply:
Though having it as a separate file does allow us to skip downloading mappings we already have.  Not sure if this is good or bad...  Good in that it saves API calls, bad in that we do not get the latest data every time.  Personally it saves me 53 calls every time I run the wizard.  Perhaps it would be better to download it every time and create a new "only fetch newly added devices" option for the wizard.

Reply:
I like that idea, a lot.  It add some complexity but also helps avoid too many API calls.  Good call out on that. We have the tension between data and limited Tuya cloud API calls we need walk.

My thought on "default to yes" is that it lets us build on the assertion that the `devices.json` file (if we choose to put it there) can contain this mapping data and be used by other apps (like the `server.py` example by @mschlenstedt , and also other home automation tools/dashboards that already use that file).  Those tools can magically unlock new capabilities because it is there. If we default to "no", I'm going to assume most new users will select the default so expanded device data (mapping) would be rare. I know, it is subtle but I do think it promotes the expansion and use of the additional attributes. For the existing TinyTuya user base, when they upgrade to the new version and run wizard, they will get a prompted default to add the mapping data as well.

What do you both think?  I see you switched us back to draft @uzlonewolf.  Are you wanting to take a try at this?

Reply:
I agree with everything said so far.  Yes, I switched back to draft so I could make these changes.

After ripping out DPMAPPINGSFILE and merging everything into devices.json, my devices.json went from 64 KiB / 1851 lines to 454 KiB / 13871 lines.  I do have a ton of devices though (127 to be exact) so this may be acceptable, not sure.  For comparison, my mappings.json was 115 KiB and 4935 lines.

Reply:
Wow! And thought I was eccentric with my 41 devices. 😄 

TBH, I don't have any problem with a 454k device file for 127 devices.  Most users are likely 10% of that but even at 4k/device, that seems reasonable for the additional functionality.  Is the difference between mappings.json due to redundant definitions (multiple same product types)?

Reply:
Yep, 127 total devices with 53 different product id's, so instead of 53 mappings there's now 127 mappings.  I have a boatload of same-model smart bulbs in track lights...